HashTagTokenizer.java

```diff
@@ -5,36 +5,78 @@ public class HashTagTokenizer {
 	public static void main(String[] args) {
 
 		String hashTag = args[0];
-		String []dictionary = readDictionary("dictionary.txt");
+		String[] dictionary = readDictionary("dictionary.txt");
 		breakHashTag(hashTag, dictionary);
 	}
 
-	public static String[] readDictionary(String fileName) {
+	public static String[] readDictionary(String fileName)
+	 {
 		String[] dictionary = new String[3000];
 
 		In in = new In(fileName);
-
-		// Your code here
+		String word = in.readLine();
+		int inedx = 0;
+		while(word!=null)
+		{
+			dictionary[inedx] = word;
+			word = in.readLine();
+			inedx++;
+		}
 
 		return dictionary;
 	}
 
-	public static boolean existInDictionary(String word, String []dictionary) {
-		// Your code here
+	public static boolean existInDictionary(String word, String []dictionary) 
+	{
+		for (int i=0;i<dictionary.length ;i++)
+		{
+			boolean same = true;
+			for(int j=0;j<dictionary[i].length();j++)//check if the value of each char is the same as the word
+			{
+				if(dictionary[i].length()!=word.length())
+				{
+					same =false;
+					break;
+				}
+				int valueofcharword = (int)(dictionary[i].charAt(j));//char of fidcetonairy
+				int valuofcharcheck = (int)(word.charAt(j));//char of the word
+				if(valueofcharword!=valuofcharcheck)
+				{
+					same = false;
+					break;
+				}
+			}
+			if(same)
+				return true;
```

Comment:
1. The boolean `same` here is unnecessary, when you found the word just return true, you don't have to mark it before return, when it found there's for sure no reason to continue any calculation.
2. You might use `word.equals` instead of validating it by your self.

Here's much more elgant solution :)
```suggestion
			if(word.equals(dictionary[I]))
				return true;
```